spec/requests/api/blueprints_spec.rb

```diff
@@ -365,6 +385,40 @@
 
       expect(blueprint.reload.bundle.dialogs).to eq([])
     end
+
+    it "publishes a blueprint without specifying a bundle name" do
+      blueprint = FactoryGirl.create(:blueprint)
+      service_template = FactoryGirl.create(:service_template)
+      service_catalog = FactoryGirl.create(:service_template_catalog)
+      service_dialog = FactoryGirl.create(:dialog_with_tab_and_group_and_field)
+      blueprint.create_bundle(:service_templates => [service_template],
+                              :service_dialog    => service_dialog,
+                              :service_catalog   => service_catalog)
+      api_basic_authorize action_identifier(:blueprints, :publish)
+
+      run_post(blueprints_url(blueprint.id), :action => "publish")
+
+      expect(blueprint.reload.published?).to be_truthy
```

Comment:
While it can be good to verify some low-level details in a highly integrated test, I think it's preferable to have exhausted all the ways in which you could verify it at the same level i.e. in the response body first. If the response says that it was successful and that it is published, since that is the only behavior that the external user sees, can we be confident that it's doing the right thing?